### DISCUSSION TITLE
Machine Laerning_1

### DISCUSSION THREAD

**Discussion by rastitiratnasari** [teamalgoritma/community#838] [Q&A] [answered]
1.  Pada meteri inclass dalam memprediksi suatu kasus terdapat tahapan mengubah kolom kategori menjadi numerik 0,1,2,,,dst.
terdapat fungsi ordinalEncoder, dan nominalEncoder. Apakah  bisa menggunakan LabelEncoder untuk menggantikan kedua fungsi tersebeut?
2. Bagaia Cara menampilan data hasil prediksi dan hasil data asli dalam satu data frame
3. Bagaimana cara mendownload hasilnya
4. Acuan menggunakan Fit() / transform()

**Comment by Yusraf002** [comment, marked as answer]
Dear @rastitiratnasari 

1. Untuk `LabelEncoder` tidak bisa menggantikan kedua fungsi tersebut. karena ditujukan untuk encoding 1 array saja dan tidak dapat mengatur urutan kelas target.  untuk penjelasan lebih detail bisa coba membaca lebih lanjut pada forum diskusi berikut: https://datascience.stackexchange.com/questions/39317/difference-between-ordinalencoder-and-labelencoder
2. Tidak perlu menggunakan pd.DataFrame untuk membuat kolom baru hasil prediksi, bisa langsung disimpan masukkan.
3. Untuk mendowload/menyimpan file data hasil prediksi, bisa gunakan fungsi `nama_data.to_csv('nama file.csv')`
4. Acuan penggunaan fit() digunakan untuk data train, sebagai contoh untuk scaling/standaraisasi akan dilakukan fit() untuk membaca pola (mean & standar deviasi) data train, dan nanti hasil perolehan fit() digunakan untuk transform() pada data train dan data test.

Apabila penjelasan ini menjawab pertanyaan Ibu, silakan klik "Mark as Answered", terima kasih

Best Regards,
Team Algoritma.

**Comment by rastitiratnasari** [comment]
Dear Mas Yusuf,

Terimakasih untuk penjelasan dan supportnya
